app/actions/bonds.ts

```diff
@@ -306,36 +419,88 @@ export const calculateUserBondDetails = (params: {
 
 export const bondTransaction = async (params: {
   value: string;
-  address: string;
   bond: Bond;
   provider: providers.JsonRpcProvider;
   slippage: number;
   onStatus: OnStatusHandler;
 }) => {
-  try {
-    const signer = params.provider.getSigner();
-    const contractAddress = getBondAddress({ bond: params.bond });
-    const contract = new ethers.Contract(
-      contractAddress,
-      Depository.abi,
-      signer
-    );
-    const calculatePremium = await contract.bondPrice();
-    const acceptedSlippage = params.slippage / 100 || 0.02; // 2%
-    const maxPremium = Math.round(calculatePremium * (1 + acceptedSlippage));
-    const valueInWei = ethers.utils.parseUnits(params.value, "ether");
-    params.onStatus("userConfirmation", "");
-    const txn = await contract.deposit(valueInWei, maxPremium, params.address);
-    params.onStatus("networkConfirmation", "");
-    await txn.wait(1);
-    params.onStatus("done", "Bond acquired successfully");
-  } catch (error: any) {
-    if (error.code === 4001) {
-      params.onStatus("error", "userRejected");
+  if (params.bond === "inverse_usdc") {
+    try {
+      const marketId = 3;
+      const acceptedSlippage = 0.0; // 20% instead of 2% bc 2% not working. 20% also not working lol
+      const signer = params.provider.getSigner();
+      const contractAddress = getBondAddress({ bond: params.bond });
+      const contract = new ethers.Contract(
+        contractAddress,
+        KlimaProV2.abi,
+        signer
+      );
+      const bondPrice = await contract.marketPrice(marketId);
+      // minimum amount to be paid out in usdc. need bondPrice in usdc
+      const minAmountOut =
+        (1 / Number(formatUnits(bondPrice, 6))) * Number(params.value) -
+        (1 / Number(formatUnits(bondPrice, 6))) *
+          Number(params.value) *
+          acceptedSlippage;
+      const formattedMinAmountOut = ethers.utils.parseUnits(
+        Number(minAmountOut.toFixed(6)).toString(),
+        "mwei"
+      );
+      // TODO add "remaining cpacity" line in view
+      // TODO check "capacity" in markets call to see if there is enough usdc in bond. returns usdc
+      params.onStatus("userConfirmation", "");
+      const notFuckedAddress = await signer.getAddress();
```

Comment:
what was wrong with params.address? plz leave a comment if we need to know
otherwise rename this var `address`

Reply:
oh, crap sorry I forgot to rename that. I have no idea what was wrong with the param. javascript was assigning it a random number instead of the address at runtime. I added a comment and change var name.